service/history/shard/context_impl.go

```diff
@@ -1985,12 +1996,14 @@ func newContext(
 	hostInfoProvider membership.HostInfoProvider,
 ) (*ContextImpl, error) {
 	hostIdentity := hostInfoProvider.HostInfo().Identity()
+	sequenceID := atomic.AddInt64(&shardContextSequenceID, 1)
 
 	lifecycleCtx, lifecycleCancel := context.WithCancel(context.Background())
 
 	shardContext := &ContextImpl{
 		state:                   contextStateInitialized,
 		shardID:                 shardID,
+		owner:                   fmt.Sprintf("%s-%v", hostIdentity, sequenceID),
```

Comment:
We can discuss what should/can be included here.

Reply:
host addr + seq num is not actually unique (maybe unique at a given time, but not over time), so I ended up appended a uuid to the end to make sure we have an unique owner. The host addr and seq num is still there for human debugging purpose.